### DISCUSSION TITLE
Poor response time of docker after loading the docker with high loads

### DISCUSSION THREAD

**Discussion by Shadow-shadown** [mediagis/nominatim-docker#523] [Q&A] [answered]
I can only load the docker image with 50 - 70 tps (transactions per second) 
I have much higher requirements in the ranger of few thousands 
Am I doing something wrong 
Or do I have to enable nginx instead of apache 
My present config is 4core cpu with 16gb ram VM ( with is small but it can be scaled up if required but I'm not sure what configuration VM to procure as I'm not aware on how to optimise to handle higher load  with good response time )
What are the configuration changes that I need to do in order to have heavily loaded and acceptable response time

**Comment by mtmail** [comment, marked as answer]
Nothing's wrong and there's no one magic switch or configuration value. You'll need to do your own benchmarking.

You need a lot more hardware. Start with 8-10x what you currently have (32 cores, 128GB) and test Apache vs nginx. Then you can finetune number of workers and clients (Apache) for example. Ideally the whole database, or at least all database indices fit into RAM. I'd be surprised if one machine will be enough, so either look for 64-128 core, 256GB RAM machine or purchase multiple.
